src/UmbracoExamine.PDF/BuilderExtensions.cs

```diff
@@ -0,0 +1,30 @@
+using Examine;
+using Microsoft.Extensions.DependencyInjection;
+using Umbraco.Cms.Core.DependencyInjection;
+using Umbraco.Cms.Core.Notifications;
+using Umbraco.Cms.Infrastructure.Examine;
+using Umbraco.Extensions;
+
+namespace UmbracoExamine.PDF
+{
+    public static class BuilderExtensions
+    {
+        public static IUmbracoBuilder AddExaminePdf(this IUmbracoBuilder builder)
+        {
+            //Register the services used to make this all work
+            builder.Services.AddUnique<IPdfTextExtractor, PdfPigTextExtractor>();
+            builder.Services.AddSingleton<PdfTextService>();
+            builder.Services.AddUnique<IPdfIndexValueSetBuilder, PdfIndexValueSetBuilder>();
+            builder.Services.AddSingleton<IIndexPopulator, PdfIndexPopulator>();
```

Comment:
I recall a discussion that this was OK, but just to flag to consider again, given we have this line in Forms...

```
builder.Services.AddSingleton<IIndexPopulator, FormsIndexPopulator>();
```

Do we have a problem here with two singleton registrations for the same interface?

Reply:
This is OKAY :) 

This thing is, we inject `IEnumeable<IIndexPopulator>` and thereby we get all of them.  

This is the differens from when you use `AddUnique`, as it would replace existing registrations.